VideoWeb/VideoWeb/ClientApp/src/app/shared/pipes/ngx-date.pipe.spec.ts

```diff
@@ -0,0 +1,21 @@
+import { NgxDatePipe } from './ngx-date.pipe';
+import { translateServiceSpy } from 'src/app/testing/mocks/mock-translation.service';
+
+describe('DatePipe', () => {
+    const translateService = translateServiceSpy;
+    const today = new Date('2021-06-16');
+
+    beforeEach(() => {
+        translateService.instant.calls.reset();
+    });
+
+    it('should return date format', () => {
+        const pipe = new NgxDatePipe(translateService);
+        expect(pipe.transform(today, 'dd MMM yyyy')).toBe('16 Jun 2021');
```

Comment:
This is really testing DatePipe. For our test we should just ensure we pass and return the correct values and not actually hit DatePipe. I can show you how to do this.

Reply:
But actually looking at it, it's quite hard to properly unit test this pipe properly as we create a new DatePipe each call. Strictly speaking we should inject a factory to create that.

Reply:
This is a more isolated version but still not perfect as we can't ensure the correct language is passed.

```suggestion
describe('DatePipe', () => {
    const translateService = translateServiceSpy;
    const testDate = new Date('2021-06-16');
    const testFormat = 'TestFormat';

    let pipe: NgxDatePipe;
    const expectedValue = 'expectedValue';

    beforeEach(() => {
        translateService.instant.calls.reset();
        pipe = new NgxDatePipe(translateService);
        spyOn(DatePipe.prototype, 'transform').and.returnValue('expectedValue');
    });

    it('should return correct value from DatePipe', () => {
        const value = pipe.transform(testDate, testFormat);
        expect(DatePipe.prototype.transform).toHaveBeenCalledWith(testDate, testFormat);
        expect(DatePipe.prototype.transform).toHaveBeenCalledTimes(1);
        expect(value).toBe(expectedValue);
    });
});
```